lib/mix/lib/mix/tasks/test.ex

```diff
@@ -228,10 +228,21 @@ defmodule Mix.Tasks.Test do
       `["test"]` if the `test` directory exists, otherwise, it defaults to `[]`.
       It is expected that all test paths contain a `test_helper.exs` file
 
-    * `:test_pattern` - a pattern to load test files. Defaults to `*_test.exs`
+    * `:test_pattern` - a pattern to find potential test files.
+      Defaults to `*.{ex,exs}`
 
-    * `:warn_test_pattern` - a pattern to match potentially misnamed test files
-      and display a warning. Defaults to `*_test.ex`
+    * `:test_load_pattern` - a regular expression to load test files.
+      Defaults to `~r/.*_test\\.exs$/`
+
+    * `:test_warn_pattern` - a regular expression to match potentially
+      misnamed test files and display a warning.
+      Defaults to `~r/^(?!.*_helper\.exs$)(?:.*_test\.ex|.*\.exs)$/`,
+      matching any file that ends in `_test.ex` or `.exs`, ignoring files
+      ending in `_helper.exs`. Warnings can be disabled by setting this to
+      `nil` or `false`.
```

Comment:
A list of filters, which can either be regexes or 1-arity functions is going to be more expressive long term, no?

Reply:
Changed it to allow strings (file paths), a regex or a one-arity function while consolidating it into `test_warn_ignore_patterns`